cmd/osquery-perf/agent.go

```diff
@@ -777,28 +777,32 @@ func (a *agent) runLoop(i int, onlyAlreadyEnrolled bool) {
 		// check if we have any scheduled queries that should be returning results
 		var results []resultLog
 		now := time.Now().Unix()
-		a.scheduledQueriesMu.Lock()
 		prevCount := a.countBuffered()
-		for queryName, query := range a.scheduledQueryData {
+
+		queryData := a.scheduledQueryData
```

Comment:
Don't you have to assign back to `a.scheduledQueryData` after the `Range`?

Reply:
Oh you're totally right 🤦

Reply:
Actually... I think we don't want to. In the unlikely case that the map was reassigned while we were iterating over it, that would cause the newly applied map to get overwritten

Reply:
Right, only thing that could happen is the `lastRun` is not updated.

So, am ok to leave as is, just leave a comment in case this happens (lastRun not being updated which means queries being executed more frequently than expected).

Reply:
Sounds good!